### DISCUSSION TITLE
REST API

### DISCUSSION THREAD

**Discussion by marcorocca1973** [DevTrevi/DgcReader#30] [Ideas]
Hi Davide
could be possible transform this library in REST service to expose an API?
Thk
Marco

**Comment by DevTrevi** [comment]
Hi @marcorocca1973 , from a technical point of view it is possible, the library is already optimized to be used in this kind of context.
When using the provided extension methods for registering the library to the ServiceCollection, all the required services are registered as singleton and can be safely used by multiple concurrent calls

If you need to implement this kind of solution, be sure to to be compliant with the current relgulations, especially in terms of privacy. It is generally forbidden to send certificates to third party services, but if you consume your API within your software solution only you should be ok
